core/ledger/kvledger/kv_ledger.go

```diff
@@ -887,6 +887,16 @@ func (l *kvLedger) GetMissingPvtDataTracker() (ledger.MissingPvtDataTracker, err
 	return l, nil
 }
 
+// CommitNotificationsChannel returns a read-only channel on which ledger sends a `CommitNotification`
+// when a block is committed. The CommitNotification contains entries for the transactions from the committed block,
+// which are not malformed, carry a legitimate TxID, and in addition, are not marked as a duplicate transaction.
+// The consumer can send a struct{} to the signal channel when the notifications are no longer needed. This will cause the
+// CommitNotifications channel to close. There is expected to be only one consumer at a time. The function returns error
+// if already a CommitNotification channel is active.
+func (l *kvLedger) CommitNotificationsChannel() (<-chan *ledger.CommitNotification, chan<- struct{}, error) {
```

Comment:
Why return a channel for the consumer to close instead of accepting the channel from the consumer? Or support passing a context that provides the same signal via `Done`?

IMO,if it's the caller's job to close the channel, the caller should provide the channel.

Reply:
Yes, agreed. Fixed in the new patch.